storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2369,7 +2369,7 @@ public void launchServer() throws Exception {
             for (ClusterMetricsConsumerExecutor exec: clusterConsumerExceutors) {
                 exec.prepare();
             }
-            
+            assertIsLeader();
             if (isLeader()) {
```

Comment:
This is a little odd.  Why do we need to be sure that we are the leader here?  Especially with code right below it checking if we are the leader or not.